v2/tools/generator/internal/conversions/property_conversions.go

```diff
@@ -195,10 +195,20 @@ func NameOfPropertyAssignmentFunction(
 //
 // <propertyBag>.Add(<propertyName>, <source>)
 //
-// For optional sources, the value is only added if non-nil
+// For optional sources, the value is only added if non-nil; if nil, we remove any existing item
 //
 // if <source> != nil {
 //     <propertyBag>.Add(<propertyName>, *<source>)
+// } else {
```

Comment:
I'm surprised that this change only shows up in 3 places in the actual generated types. Is that expected? I would have thought we had more optional sources than that.

Reply:
It's not just that the source is optional - it's that the field is present in one storage version of the resource and not in another storage version; this is pretty rare at the moment because we only have one version of most resources.